crates/flowctl/src/raw/oauth.rs

```diff
@@ -0,0 +1,273 @@
+        .from_env_lossy();
+
+    let guard = tracing::subscriber::set_default(
+        tracing_subscriber::fmt::fmt()
+            .with_env_filter(env_filter)
+            .compact()
+            .without_time()
+            .with_target(false)
+            .with_writer(std::io::stderr)
+            .finish(),
+    );
+    // END TESTING
+    let source = build::arg_source_to_url(source, false)?;
+    let mut sources = local_specs::surface_errors(local_specs::load(&source).await.into_result())?;
+
+    // Identify the capture to inspect.
+    // TODO (jshearer): materialization support
+    let needle = if let Some(needle) = task {
+        needle.as_str()
+    } else if sources.captures.len() == 1 {
+        sources.captures.first().unwrap().capture.as_str()
+    } else if sources.captures.is_empty() {
+        anyhow::bail!("sourced specification files do not contain any captures");
+    } else {
+        anyhow::bail!("sourced specification files contain multiple captures. Use --capture to identify a specific one");
+    };
+
+    let capture = match sources
+        .captures
+        .binary_search_by_key(&needle, |c| c.capture.as_str())
+    {
+        Ok(index) => &mut sources.captures[index],
+        Err(_) => anyhow::bail!("could not find the capture {needle}"),
+    };
+
+    tracing::info!(
+        task_name = capture.capture.as_str(),
+        "Performing oauth flow on task"
+    );
+
+    let spec_req = match &capture.spec.endpoint {
+        models::CaptureEndpoint::Connector(config) => capture::request::Spec {
+            connector_type: flow::capture_spec::ConnectorType::Image as i32,
+            config_json: serde_json::to_string(&config).unwrap(),
+        },
+        models::CaptureEndpoint::Local(config) => capture::request::Spec {
+            connector_type: flow::capture_spec::ConnectorType::Local as i32,
+            config_json: serde_json::to_string(config).unwrap(),
+        },
+    };
+    // Get the task spec's oauth field
+    let mut spec_req = capture::Request {
+        spec: Some(spec_req),
+        ..Default::default()
+    };
+
+    if let Some(log_level) = capture
+        .spec
+        .shards
+        .log_level
+        .as_ref()
+        .and_then(|s| ops::LogLevel::from_str_name(s))
+    {
+        spec_req.set_internal_log_level(log_level);
+    }
+
+    let spec_response = runtime::Runtime::new(
+        true, // All local.
+        network.clone(),
+        ops::tracing_log_handler,
+        None,
+        format!("spec/{}", capture.capture),
+    )
+    .unary_capture(spec_req, build::CONNECTOR_TIMEOUT)
+    .await
+    .map_err(crate::status_to_anyhow)?
+    .spec
+    .context("connector didn't send expected Spec response")?;
+
+    let oauth_spec = spec_response
+        .oauth2
+        .expect("Connector did not return an oauth config");
+
+    // TODO (jshearer): Validate endpoint config against spec.config_schema_json
```

Comment:
Should this be done now, or just something that might be nice?

Reply:
Done